src/update.ts

```diff
@@ -268,14 +268,22 @@ export class Updater {
     return current === updated
   }
 
-  private async determineChannel(): Promise<string> {
+  private async determineChannel(version?:string): Promise<string> {
     const channelPath = path.join(this.config.dataDir, 'channel')
-    if (fs.existsSync(channelPath)) {
-      const channel = await fs.readFile(channelPath, 'utf8')
-      return String(channel).trim()
-    }
 
-    return this.config.channel || 'stable'
+    const channel = fs.existsSync(channelPath) ? (await fs.readFile(channelPath, 'utf8')).trim() : 'stable'
+
+    try {
+      const {body} = await HTTP.get<{'dist-tags':Record<string, string>}>(`https://registry.npmjs.org/${this.config.pjson.name}/`)
```

Comment:
I'm glad I got another look at this.  I thought of something else.

There's a config and and env that oclif uses.  see
https://github.com/oclif/core/blob/0c34a9d3d148f260384c0c8d3b7aa5ab974056ad/src/config/config.ts#L165
https://github.com/oclif/plugin-plugins/blob/ae60398dcf28fbbd65240c1979dceaadc86c35b4/src/yarn.ts#L61
https://github.com/oclif/oclif.github.io/blob/cb01783492cf177cc89be1717a39f5f396424fa0/docs/config.md#L34

I think this should support those.